### PR TITLE
chore(ci): set-up a workflow to use the RTX 4090

### PR DESCRIPTION
**Pull request by IceTDrinker**
Add an RTX 4090 CI workflow triggered on label

### CODE REVIEW

File: .github/workflows/aws_tfhe_gpu_4090_tests.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Compile and test tfhe-cuda-backend on an AWS instance
```
Comment (soonum) [inline .github/workflows/aws_tfhe_gpu_4090_tests.yml:1]: Not on an AWS instance. :wink:

File: .github/workflows/aws_tfhe_gpu_4090_tests.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+on:
+  # Allows you to run this workflow manually from the Actions tab as an alternative.
+  workflow_dispatch:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  cuda-tests-linux:
+    name: CUDA tests
```
Comment (soonum) [inline .github/workflows/aws_tfhe_gpu_4090_tests.yml:22]: We use the same name in the AWS version. Could we have a way to distinguish between the two flavors ?
Maybe something like `CUDA tests (RTX 4090)`

File: .github/workflows/aws_tfhe_gpu_4090_tests.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          labels: 4090_test
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Slack Notification
+        if: ${{ always() }}
+        continue-on-error: true
+        uses: rtCamp/action-slack-notify@b24d75fe0e728a4bf9fc42ee217caa686d141ee8
+        env:
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_MESSAGE: "CUDA AWS tests finished with status: ${{ job.status }}. (${{ env.ACTION_RUN_URL }})"
```
Comment (soonum) [inline .github/workflows/aws_tfhe_gpu_4090_tests.yml:77]: Not `CUDA AWS...`

File: .github/workflows/aws_tfhe_gpu_4090_tests.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      - cuda-tests-linux
+    if: ${{ always() }}
+    steps:
+      - uses: actions-ecosystem/action-remove-labels@2ce5d41b4b6aa8503e285553f75ed56e0a40bae0
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          labels: 4090_test
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Slack Notification
```
Comment (soonum) [inline .github/workflows/aws_tfhe_gpu_4090_tests.yml:71]: The location of the notification step seems a bit odd since it's inside `remove_label` job.

Comment (IceTDrinker) [reply .github/workflows/aws_tfhe_gpu_4090_tests.yml:71]: I copied the M1 mac workflow and it looks to work ok 🤔

**Review by soonum — APPROVED** [review]
Now we're talking ! :smiley:

**Pull request merged**